coala_quickstart/green_mode/file_aggregator.py

```diff
@@ -0,0 +1,81 @@
+import os
+
+from coala_quickstart.green_mode.find_globs import (
+    find_globs_from_files,
+    )
+
+
+class Node:
+    """
+    Tree data-structure that has each directory as a node,
+    each subdirectory as children and files as leaves.
+    """
+
+    def __init__(self, name, parent, children=[]):
+        self.name = name
+        self.parent = parent
+        self.children = children
+
+    def add_children(self, children):
+        self.children.append(children)
+
+    def search_name(self, name):
+        if name == self.name:  # pragma: no cover
+            return self
+        for child in self.children:
+            if child.name == name:
+                return child
+        for child in self.children:
+            found_node = child.search_name(name)
+            if found_node is not None:  # pragma: no cover
+                return found_node
+        return None
+
+    def get_files(self, initials):
+        for child in self.children:
+            name = initials + os.sep + child.name
+            # FIXME: What happens in case of empty directories?
```

Comment:
Return `None`? Or is there any peculiar case that I'm missing?

Reply:
No, currently in case of empty directories it detects it as files. A commit attaching an attribute to the Node object when the tree is initially constructed would solve it (as there it actually checks whether the item is a file and only then adds it as a leaf)